process/peer/process_test.go

```diff
@@ -1014,6 +1029,37 @@ func TestValidatorStatisticsProcessor_UpdatePeerState_DecreasesMissedMetaBlock_P
 	assert.Equal(t, uint32(1), missedValidator.DecreaseValidatorSuccessRateValue)
 }
 
+func TestValidatorStatisticsProcessor_UpdatePeerStateV3(t *testing.T) {
+	t.Parallel()
+
+	consensusGroup := make(map[string][]nodesCoordinator.Validator)
+
+	arguments := createUpdateTestArgs(consensusGroup)
+	arguments.EnableEpochsHandler = &enableEpochsHandlerMock.EnableEpochsHandlerStub{
+		IsFlagEnabledInEpochCalled: func(flag core.EnableEpochFlag, epoch uint32) bool {
+			return true
+		},
+	}
+	validator := shardingMocks.NewValidatorMock([]byte("pk1"), 1, 1)
+	arguments.NodesCoordinator = &shardingMocks.NodesCoordinatorStub{
+		ComputeConsensusGroupCalled: func(randomness []byte, round uint64, shardId uint32, epoch uint32) (leader nodesCoordinator.Validator, validatorsGroup []nodesCoordinator.Validator, err error) {
+			return validator, []nodesCoordinator.Validator{validator}, nil
+		},
+	}
+	validatorStatistics, _ := peer.NewValidatorStatisticsProcessor(arguments)
+
+	cache := createMockCache()
+	prevHeader, header := generateTestMetaBlockV3Headers(cache)
+	prevHeader.EpochStart.LastFinalizedHeaders = []block.EpochStartShardData{{ShardID: 0}}
+
+	metaExecResult := &block.MetaExecutionResult{
+		AccumulatedFees: big.NewInt(1),
+		DeveloperFees:   big.NewInt(1),
+	}
+	_, err := validatorStatistics.UpdatePeerStateV3(header, cache, metaExecResult)
```

Comment:
[optional]
Would've been more complete to actually test the same behaviour as in `UpdatePeerState`, with different input. (like take one existing test and check same result, different inputs)

Reply:
agree, will be done on a different PR